google-cloud-spanner/lib/google/cloud/spanner/snapshot.rb

```diff
@@ -288,14 +288,15 @@ def timestamp
         #   end
         #
         def execute_query sql, params: nil, types: nil, query_options: nil,
-                          call_options: nil
+                          call_options: nil, directed_read_options: nil
```

Comment:
I have not looked at the PR yet, but noticed that partition read and query cases are not handled. Can you please take a look?

Reply:
Thats a great catch thanks for pointing it out, I have updated to add support for batch_client PTAL 👍